src/ConstructiveSolidGeometry/LinePrimitives/Line.jl

```diff
@@ -0,0 +1,74 @@
+struct Line{T,TP,TL} <: AbstractLinePrimitive{T}
+    p1::TP
+    p2::TP
+    linetype::TL
+    function Line( ::Type{T},
+                   p1::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   p2::Union{CartesianPoint{T}, PlanarPoint{T}},
+                   linetype::Union{Val{:inf}, Val{:ray}, Val{:seg}}) where {T}
+        new{T,typeof(p1),typeof(linetype)}(p1, p2)
+    end
```

Comment:
As `p1` **MUST** always be the same type as `p2`:
```julia
function Line( ::Type{T},
                p1::PT,
                p2::PT,
                linetype::Union{Val{:inf}, Val{:ray}, Val{:seg}}
              ) where {T, PT <: Union{CartesianPoint{T}, PlanarPoint{T}}}
    new{T,typeof(p1),typeof(linetype)}(p1, p2)
end
```

Same below for the convenience functions `Line`, `Ray` and `LineSegment`.